go-controller/pkg/ovn/pods.go

```diff
@@ -615,139 +570,44 @@ func (oc *Controller) getPortAddresses(nodeName, portName string) (net.HardwareA
 	return podMac, podIPNets, nil
 }
 
-func (oc *Controller) addPodExternalGW(pod *kapi.Pod) error {
-	type Network struct {
-		Name string
-		Ips  []string
-	}
-	routingNamespaceAnnotation := pod.Annotations[routingNamespaceAnnotation]
-	if routingNamespaceAnnotation == "" {
-		return nil
-	}
-	klog.Infof("External gateway pod: %s, detected for namespace(s) %s", pod.Name, routingNamespaceAnnotation)
-	var foundGws []net.IP
-	if routingNetworkAnnotation == "" && pod.Spec.HostNetwork {
-		for _, podIP := range pod.Status.PodIPs {
-			ip := net.ParseIP(podIP.IP)
-			if ip != nil {
-				foundGws = append(foundGws, ip)
-			}
-		}
-	} else if routingNetworkAnnotation != "" {
-		var multusNetworks []Network
-		err := json.Unmarshal([]byte(pod.ObjectMeta.Annotations["k8s.v1.cni.cncf.io/network-status"]), &multusNetworks)
-		if err != nil {
-			return fmt.Errorf("unable to unmarshall annotation k8s.v1.cni.cncf.io/network-status on pod %s: %v", pod.Name, err)
-		}
-		for _, multusNetwork := range multusNetworks {
-			if multusNetwork.Name == routingNetworkAnnotation {
-				for _, gwIP := range multusNetwork.Ips {
-					ip := net.ParseIP(gwIP)
-					if ip != nil {
-						foundGws = append(foundGws, ip)
-					}
-				}
-			}
-		}
-	} else {
-		klog.Errorf("Ignoring pod %s as an external gateway candidate. Invalid combination "+
-			"of host network: %t and routing-network annotation: %s", pod.Name, pod.Spec.HostNetwork, routingNetworkAnnotation)
-		return nil
-	}
-
-	// if we found any gateways then we need to update current pods routing in the relevant namespace
-	if len(foundGws) == 0 {
-		klog.Warningf("No valid gateway IPs found for requested external gateway pod: %s", pod.Name)
-		return nil
-	}
-
-	for _, namespace := range strings.Split(routingNamespaceAnnotation, ",") {
-		nsInfo, err := oc.waitForNamespaceLocked(namespace)
-		if err != nil {
-			return err
-		}
-		defer nsInfo.Unlock()
-		nsInfo.routingExternalPodGWs[pod.Name] = foundGws
-		existingPods, err := oc.watchFactory.GetPods(namespace)
+// deletePerPodGRSNAT removes per pod SNAT rules that are applied to the GR where the pod resides if
+// there are no gateways
+func (oc *Controller) deletePerPodGRSNAT(node string, podIPNets []*net.IPNet) {
```

Comment:
This stuff also seems like it could move to egressgw?

Reply:
sure I can move it there. I tried to add unit tests for the GRSNAT stuff but it was a real pain because we have to have nodes added for this to work. I started down that path and it ended up a mess, so skipping coverage on those for now.